modules/serverless-apiserversource.adoc

```diff
@@ -0,0 +1,231 @@
+
+* For steps that use `kn`, you will need to have the `kn` CLI installed. For more information, see the documentation on xref:../../serverless/installing-kn.adoc#installing-kn[Installing the Knative CLI].
+
+.Procedure
+
+. Create a Service Account for the ApiServerSource.
++
+You can do this by creating a file named `serviceaccount.yaml` and copying the following sample code into it:
++
+----
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: events-sa
+  namespace: default
+
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: event-watcher
+rules:
+  - apiGroups:
+      - ""
+    resources:
+      - events
+    verbs:
+      - get
+      - list
+      - watch
+
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: k8s-ra-event-watcher
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: event-watcher
+subjects:
+  - kind: ServiceAccount
+    name: events-sa
+    namespace: default
+----
++
+[NOTE]
+====
+If you want to re-use an existing Service Account with the appropriate permissions, you must modify the `serviceaccount.yaml` for that Service Account.
+====
++
+Create the service account by entering the following command:
++
+----
+$ oc apply --filename serviceaccount.yaml
+----
+
+. Create an ApiServerSource event source.
++
+You can do this by creating a file named `k8s-events.yaml` and copying the following sample code into it:
++
+----
+apiVersion: sources.knative.dev/v1alpha1
+kind: ApiServerSource
+metadata:
+  name: testevents
+  namespace: default
+spec:
+  serviceAccountName: events-sa
+  mode: Resource
+  resources:
+    - apiVersion: v1
+      kind: Event
+  sink:
+    ref:
+      apiVersion: eventing.knative.dev/v1beta1
+      kind: Broker
+      name: default
+----
++
+[NOTE]
+====
+If you want to consume events from a different namespace or use a different Service Account, you need to modify the sample `k8s-events.yaml` file accordingly.
+====
++
+Create the event source by entering the following command:
++
+----
+$ oc apply --filename k8s-events.yaml
+----
+
+. To check that the ApiServerSource is set up correctly, create a Knative service that dumps incoming messages to its log.
++
+Copy the following sample YAML into a file named `service.yaml`:
++
+----
+apiVersion: serving.knative.dev/v1
+kind: Service
+metadata:
+  name: event-display
+  namespace: default
+spec:
+  template:
+    spec:
+      containers:
+        - image: quay.io/openshift-knative/knative-eventing-sources-event-display
+----
++
+Create the service by entering the following command:
++
+----
+$ oc apply --filename service.yaml
+----
+
+. Create a trigger from the `default` broker to filter events to the service created in the previous step.
++
+You can create the trigger by entering the following command:
++
+----
+$ kn trigger create foo --broker default --sink svc:event-display \
```

Comment:
Do you really want to call the trigger "foo" ? I would suggest to change the name to "event-display-trigger"